api4/role.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package api4
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/model"
+)
+
+func (api *API) InitRole() {
+	api.BaseRoutes.Roles.Handle("/{role_id:[A-Za-z0-9]+}", api.ApiSessionRequiredTrustRequester(getRole)).Methods("GET")
+	api.BaseRoutes.Roles.Handle("/name/{role_name:[a-z0-9_]+}", api.ApiSessionRequiredTrustRequester(getRoleByName)).Methods("GET")
+	api.BaseRoutes.Roles.Handle("/names", api.ApiSessionRequiredTrustRequester(getRolesByNames)).Methods("POST")
+	api.BaseRoutes.Roles.Handle("/{role_id:[A-Za-z0-9]+}/patch", api.ApiSessionRequired(patchRole)).Methods("PUT")
```

Comment:
Can we remove the verb from the url here? The fact that it's a PUT method implies that it is patching an existing role.

Reply:
Consistency with the other patch API endpoints.

Reply:
PUT actually doesn't necessarily imply a patch, there is a specific verb PATCH for that. We avoided using that because it's not present in older implementations of HTTP and we didn't want to break any old load balancers customers might be using. So in our API `PUT /resource/{id}` is a full update while `PUT /resource/{id}/patch` is a patch